cmd/agentctl/cli/config.go

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2019 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ligato/cn-infra/config"
+	"github.com/ligato/cn-infra/logging"
+)
+
+const (
+	configFileDir  = ".agentctl"
+	configFileName = "all.conf"
+)
+
+// ConfigFile represents info from ~/.agentctl/all.conf.
+type ConfigFile struct {
```

Comment:
I agree with comment from @rastislavszabo https://github.com/ligato/vpp-agent/pull/1534#issuecomment-544927983
> 
> 
> Anyway, it would be useful if etcd endpoints could be passed via config file as well. Maybe even if **any agentctl argument (e.g. service label) could be passed via config file**?

This config file definition (rather `Config`) should be summarizing all available configuration options that AgentCTL accepts/requires. There should be a `DefaultConfig()` function that would set the default values which should be used on line 56, then some of the options can get overridden by reading the config file contents, after that env vars and lastly the program arguments should be checked to [properly handle precedence order](https://github.com/spf13/viper#why-viper):

- default config
- config file (possibly system and then local)
- env vars
- program arguments

The Go library [viper](https://godoc.org/github.com/spf13/viper) might be really helpful here and provide simpler way to make this all work automatically.

Reply:
> a complete configuration for the AgentCTL

Sure, complete configuration sounds good. Will work on that too.

> The Go library viper might be really helpful

yeah, tab with Viper is opened in my browser from the beginning of work on this task, but for now I'm fighting for better TLS (I don't like that code from cn-infra). Definitely will look into Viper, but a little bit later, when I'll be working on a complete configuration and that order of different ways of configurating AgentCTL, I think.